staging/src/k8s.io/apiserver/pkg/admission/plugin/validatingadmissionpolicy/compiler.go

```diff
@@ -193,22 +241,20 @@ func CompileValidatingPolicyExpression(validationExpression string, hasParams bo
 			},
 		}
 	}
-	if ast.OutputType() != cel.BoolType {
+	ast, issues := env.Check(parsedAst)
+	if issues != nil {
 		return CompilationResult{
 			Error: &apiservercel.Error{
 				Type:   apiservercel.ErrorTypeInvalid,
-				Detail: "cel expression must evaluate to a bool",
+				Detail: "compilation failed: " + issues.String(),
```

Comment:
nit: Should we state it more specifically on type checking failure?

Reply:
This is a diff issue. I did not change these lines😅.